raspberry pi runtime error

## Description

I compiled a docker file image (v1.5.0) for my raspberry pi (Linux raspberrypi 5.10.63-v7+ #1459 SMP Wed Oct 6 16:41:10 BST 2021 armv7l GNU/Linux) but pactus daemon exit with the following error:

```
panic: runtime error: index out of range [4] with length 4

goroutine 1 [running]:
github.com/kilic/bls12-381.(*Fr).fromBig(0x3d19ad4, 0x3d194fc)
	github.com/kilic/bls12-381@v0.1.0/fr.go:84 +0x274
github.com/kilic/bls12-381.(*Fr).fromBytes(0x3d19ad4, {0x4086480, 0x20, 0x40})
	github.com/kilic/bls12-381@v0.1.0/fr.go:69 +0x88
github.com/kilic/bls12-381.(*Fr).FromBytes(...)
	github.com/kilic/bls12-381@v0.1.0/fr.go:57
github.com/pactus-project/pactus/crypto/bls/hdkeychain.(*ExtendedKey).Derive(0x3ff06c0, 0x8000305d)
	github.com/pactus-project/pactus/crypto/bls/hdkeychain/extendedkey.go:232 +0x7b8
github.com/pactus-project/pactus/crypto/bls/hdkeychain.(*ExtendedKey).DerivePath(0x3ff06c0, {0x40441b0, 0x4, 0x4})
	github.com/pactus-project/pactus/crypto/bls/hdkeychain/extendedkey.go:107 +0x4c
github.com/pactus-project/pactus/wallet/vault.(*Vault).deriveBLSPrivateKey(0x3f4e040, {0x4086200, 0x40, 0x40}, {0x40441b0, 0x4, 0x4})
	github.com/pactus-project/pactus/wallet/vault/vault.go:623 +0x64
github.com/pactus-project/pactus/wallet/vault.(*Vault).PrivateKeys(0x3f4e040, {0x7ee70f5a, 0x8}, {0x3f4e0c0, 0x7, 0x7})
	github.com/pactus-project/pactus/wallet/vault/vault.go:425 +0x3d4
github.com/pactus-project/pactus/wallet.(*Wallet).PrivateKeys(...)
	github.com/pactus-project/pactus/wallet/wallet.go:406
github.com/pactus-project/pactus/cmd.MakeValidatorKey(0x4116420, {0x3ef20f8, 0x7, 0x7}, 0x410fe2c)
	github.com/pactus-project/pactus/cmd/cmd.go:625 +0x290
github.com/pactus-project/pactus/cmd.StartNode({0x408ec40, 0xc}, 0x410fe2c)
	github.com/pactus-project/pactus/cmd/cmd.go:406 +0x204
main.buildStartCmd.func1(0x3fccc08, {0x4084fb0, 0x0, 0x2})
	github.com/pactus-project/pactus/cmd/daemon/start.go:60 +0x1b0
github.com/spf13/cobra.(*Command).execute(0x3fccc08, {0x4084f90, 0x2, 0x2})
	github.com/spf13/cobra@v1.8.1/command.go:989 +0x9b0
github.com/spf13/cobra.(*Command).ExecuteC(0x3fcc608)
	github.com/spf13/cobra@v1.8.1/command.go:1117 +0x44c
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
	github.com/pactus-project/pactus/cmd/daemon/main.go:29 +0xd8

```

### How To Reproduce

```
docker build -t pactus:rasp .
docker run -it -d -v ~/pactus:/root/pactus -p 21888:21888 -p 21888:21888/udp -p 50051:50051 -p 8080:8080 --name pactus pactus:rasp pactus-daemon start --password <password>
```

--- a/go.mod
+++ b/go.mod
@@ -19,7 +19,7 @@ require (
 	github.com/hashicorp/golang-lru/v2 v2.0.7
 	github.com/inancgumus/screen v0.0.0-20190314163918-06e984b86ed3
 	github.com/k0kubun/go-ansi v0.0.0-20180517002512-3bf9e2903213
-	github.com/kilic/bls12-381 v0.1.0
+	github.com/kilic/bls12-381 v0.1.1-0.20210503002446-7b7597926c69
 	github.com/libp2p/go-libp2p v0.37.0
 	github.com/libp2p/go-libp2p-kad-dht v0.28.1
 	github.com/libp2p/go-libp2p-pubsub v0.12.0

--- a/state/state.go
+++ b/state/state.go
@@ -184,7 +184,7 @@ func (st *state) loadMerkels() {
 			panic(fmt.Sprintf(
 				"Account number is out of range: %v >= %v", acc.Number(), totalAccount))
 		}
-		st.accountMerkle.SetHash(int(acc.Number()), acc.Hash())
+		st.accountMerkle.SetHash(acc.Number(), acc.Hash())
 
 		return false
 	})
@@ -196,7 +196,7 @@ func (st *state) loadMerkels() {
 			panic(fmt.Sprintf(
 				"Validator number is out of range: %v >= %v", val.Number(), totalValidator))
 		}
-		st.validatorMerkle.SetHash(int(val.Number()), val.Hash())
+		st.validatorMerkle.SetHash(val.Number(), val.Hash())
 
 		return false
 	})
@@ -529,14 +529,14 @@ func (st *state) commitSandbox(sbx sandbox.Sandbox, round int16) {
 	sbx.IterateAccounts(func(addr crypto.Address, acc *account.Account, updated bool) {
 		if updated {
 			st.store.UpdateAccount(addr, acc)
-			st.accountMerkle.SetHash(int(acc.Number()), acc.Hash())
+			st.accountMerkle.SetHash(acc.Number(), acc.Hash())
 		}
 	})
 
 	sbx.IterateValidators(func(val *validator.Validator, updated bool, _ bool) {
 		if updated {
 			st.store.UpdateValidator(val)
-			st.validatorMerkle.SetHash(int(val.Number()), val.Hash())
+			st.validatorMerkle.SetHash(val.Number(), val.Hash())
 		}
 	})
 

--- a/util/persistentmerkle/merkle.go
+++ b/util/persistentmerkle/merkle.go
@@ -7,14 +7,14 @@ import (
 )
 
 type Tree struct {
-	nodes     map[int]*node
-	maxWidth  int
-	maxHeight int
+	nodes     map[uint32]*node
+	maxWidth  int32
+	maxHeight int32
 }
 
 type node struct {
-	width  int
-	height int
+	width  int32
+	height int32
 	hash   *hash.Hash
 }
 
@@ -24,30 +24,30 @@ type node struct {
 // +-+---+
 // h: height
 // w: width
-func nodeID(width, height int) int {
-	return ((height & 0xff) << 24) | (width & 0xffffff)
+func nodeID(width, height int32) uint32 {
+	return (uint32(height&0xff) << 24) | uint32(width&0xffffff)
 }
 
 func New() *Tree {
 	return &Tree{
-		nodes: make(map[int]*node),
+		nodes: make(map[uint32]*node),
 	}
 }
 
-func (*Tree) createNode(width, height int) *node {
+func (*Tree) createNode(width, height int32) *node {
 	return &node{
 		width:  width,
 		height: height,
 	}
 }
 
-func (t *Tree) getNode(width, height int) *node {
+func (t *Tree) getNode(width, height int32) *node {
 	id := nodeID(width, height)
 
 	return t.nodes[id]
 }
 
-func (t *Tree) getOrCreateNode(width, height int) *node {
+func (t *Tree) getOrCreateNode(width, height int32) *node {
 	id := nodeID(width, height)
 	node, ok := t.nodes[id]
 	if !ok {
@@ -58,36 +58,36 @@ func (t *Tree) getOrCreateNode(width, height int) *node {
 	return node
 }
 
-func (t *Tree) invalidateNode(width, height int) {
+func (t *Tree) invalidateNode(width, height int32) {
 	n := t.getOrCreateNode(width, height)
 	n.hash = nil
 }
 
-func (t *Tree) recalculateHeight(maxWidth int) {
+func (t *Tree) recalculateHeight(maxWidth int32) {
 	if maxWidth > t.maxWidth {
 		t.maxWidth = maxWidth
 
 		maxHeight := math.Log2(float64(maxWidth))
 		if math.Remainder(maxHeight, 1.0) != 0 {
-			t.maxHeight = int(math.Trunc(maxHeight)) + 2
+			t.maxHeight = int32(math.Trunc(maxHeight)) + 2
 		} else {
-			t.maxHeight = int(math.Trunc(maxHeight)) + 1
+			t.maxHeight = int32(math.Trunc(maxHeight)) + 1
 		}
 	}
 }
 
-func (t *Tree) SetData(leaf int, data []byte) {
+func (t *Tree) SetData(leaf int32, data []byte) {
 	t.SetHash(leaf, hash.CalcHash(data))
 }
 
-func (t *Tree) SetHash(leaf int, h hash.Hash) {
+func (t *Tree) SetHash(leaf int32, h hash.Hash) {
 	t.recalculateHeight(leaf + 1)
 
 	node := t.getOrCreateNode(leaf, 0)
 	node.hash = &h
 
 	w := leaf / 2
-	for h := 1; h < t.maxHeight; h++ {
+	for h := int32(1); h < t.maxHeight; h++ {
 		t.invalidateNode(w, h)
 		w /= 2
 	}
@@ -97,7 +97,7 @@ func (t *Tree) Root() hash.Hash {
 	return t.nodeHash(0, t.maxHeight-1)
 }
 
-func (t *Tree) nodeHash(width, height int) hash.Hash {
+func (t *Tree) nodeHash(width, height int32) hash.Hash {
 	node := t.getNode(width, height)
 	if node == nil {
 		node = t.getNode(width-1, height)

--- a/util/persistentmerkle/merkle_test.go
+++ b/util/persistentmerkle/merkle_test.go
@@ -8,38 +8,38 @@ import (
 )
 
 func TestNodeID(t *testing.T) {
-	assert.Equal(t, 0x00000000, nodeID(0, 0))
-	assert.Equal(t, 0x01000000, nodeID(0, 1))
-	assert.Equal(t, 0x00000001, nodeID(1, 0))
-	assert.Equal(t, 0x01000001, nodeID(1, 1))
-	assert.Equal(t, 0xffffffff, nodeID(0xffffff, 0xff))
-	assert.Equal(t, 0x00ffffff, nodeID(0xffffff, 0x00))
-	assert.Equal(t, 0x77ff00ff, nodeID(0xff00ff, 0x77))
+	assert.Equal(t, uint32(0x00000000), nodeID(0, 0))
+	assert.Equal(t, uint32(0x01000000), nodeID(0, 1))
+	assert.Equal(t, uint32(0x00000001), nodeID(1, 0))
+	assert.Equal(t, uint32(0x01000001), nodeID(1, 1))
+	assert.Equal(t, uint32(0xffffffff), nodeID(0xffffff, 0xff))
+	assert.Equal(t, uint32(0x00ffffff), nodeID(0xffffff, 0x00))
+	assert.Equal(t, uint32(0x77ff00ff), nodeID(0xff00ff, 0x77))
 }
 
 func TestCalculateHeight(t *testing.T) {
 	tree := New()
 
 	tree.recalculateHeight(0)
-	assert.Equal(t, 0, tree.maxHeight)
+	assert.Equal(t, int32(0), tree.maxHeight)
 
 	tree.recalculateHeight(1)
-	assert.Equal(t, 1, tree.maxHeight)
+	assert.Equal(t, int32(1), tree.maxHeight)
 
 	tree.recalculateHeight(2)
-	assert.Equal(t, 2, tree.maxHeight)
+	assert.Equal(t, int32(2), tree.maxHeight)
 
 	tree.recalculateHeight(4)
-	assert.Equal(t, 3, tree.maxHeight)
+	assert.Equal(t, int32(3), tree.maxHeight)
 
 	tree.recalculateHeight(5)
-	assert.Equal(t, 4, tree.maxHeight)
+	assert.Equal(t, int32(4), tree.maxHeight)
 
 	tree.recalculateHeight(8)
-	assert.Equal(t, 4, tree.maxHeight)
+	assert.Equal(t, int32(4), tree.maxHeight)
 
 	tree.recalculateHeight(9)
-	assert.Equal(t, 5, tree.maxHeight)
+	assert.Equal(t, int32(5), tree.maxHeight)
 }
 
 func TestMerkleTree(t *testing.T) {
@@ -79,7 +79,7 @@ func TestMerkleTree(t *testing.T) {
 	}
 
 	for i, d := range data {
-		tree.SetData(i, []byte(d))
+		tree.SetData(int32(i), []byte(d))
 		expected, _ := hex.DecodeString(roots[i])
 		assert.Equal(t, expected, tree.Root().Bytes(), "Root %d not matched", i)
 	}

